library/std/src/sys/uefi/os.rs

```diff
@@ -0,0 +1,357 @@
+use super::{
+    common::{self, status_to_io_error},
+    unsupported,
+};
+use crate::ffi::{OsStr, OsString};
+use crate::fmt;
+use crate::io;
+use crate::marker::PhantomData;
+use crate::os::uefi;
+use crate::path::{self, PathBuf};
+use crate::{error::Error as StdError, os::uefi::ffi::OsStringExt};
+
+// Return EFI_ABORTED as Status
+pub fn errno() -> i32 {
+    r_efi::efi::Status::SUCCESS.as_usize() as i32
+}
+
+pub fn error_string(errno: i32) -> String {
+    let r = r_efi::efi::Status::from_usize(errno as usize);
+    status_to_io_error(r).to_string()
+}
+
+// Implemented using EFI_LOADED_IMAGE_DEVICE_PATH_PROTOCOL
+pub fn getcwd() -> io::Result<PathBuf> {
+    let mut p = current_exe()?;
+    p.pop();
+    Ok(p)
```

Comment:
The UEFI shell has a function for this (`GetCurDir`) which probably should be tried first here.

Reply:
cc @dvdhrm 

I'm a bit reluctant to use anything from UEFI Shell Protocol. From my interactions with edk2 community, in most cases, UEFI shell is not going to be available. But since many applications using std will be Applications and not drivers, maybe using UEFI shell stuff might not be a problem?